.github/workflows/test-zkvm.yml

```diff
@@ -63,77 +63,68 @@ jobs:
         id: changed_files
         uses: tj-actions/changed-files@v46
         with:
-          files_yaml: |
-            base:
-              - docker/base/Dockerfile.base
-            base_zkvm:
-              - docker/${{ inputs.zkvm }}/Dockerfile
-              - scripts/sdk_installers/install_${{ inputs.zkvm }}_sdk.sh
-       
+          files: |
+            docker/base/Dockerfile.base
+            docker/${{ inputs.zkvm }}/Dockerfile
+            scripts/sdk_installers/install_${{ inputs.zkvm }}_sdk.sh
+
       - name: Get image version and tags of ere-base and ere-base-${{ inputs.zkvm }}
-        id: inspect_image
+        id: image_tag
         run: |
           ZKVM_CRATE_VERSION=$(cargo metadata --format-version=1 --no-deps | jq -r '.packages[] | select(.name == "ere-${{ inputs.zkvm }}") | .version')
+
           GIT_REV="${{ github.sha }}"
+          if [ "${{ github.event_name }}" == "pull_request" ] && [ "${{ steps.changed_files.outputs.any_changed }}" == "false" ]; then
+            CACHED_GIT_REV="${{ github.event.pull_request.base.sha }}"
+          else
+            CACHED_GIT_REV="$GIT_REV"
+          fi
+
```

Comment:
If it is a PR and there is no changes related to the base images, we use the base images built in the PR base to run the latter clippy/test.

Because image will always be built when push to `master`, so if the PR base is `master` (and the CI is finished), the images get reused. If the PR base is not `master` or the `master` CI hasn't finished, the image will be not found, then the latter clippy/test will fallback to locally build the images.